plugins/default/lib/mongodb_health.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+import argparse
+import json
+import sys
+import traceback
+from collections import Counter
+
+import nagios
+import openshift
+
+
+def enum(*sequential, **named):
+    enums = dict(zip(sequential, range(len(sequential))), **named)
+    reverse = dict((value, key) for key, value in enums.iteritems())
+    enums['reverse_mapping'] = reverse
+    return type('Enum', (), enums)
+
+# MongoDB replset status codes:
+REPLSET_STATUS = enum(
+    STARTUP=0,
+    PRIMARY=1,
+    SECONDARY=2,
+    RECOVERING=3,
+    FATAL=4,
+    STARTUP2=5,
+    UNKNOWN=6,
+    ARBITER=7,
+    DOWN=8,
+    ROLLBACK=9,
+    REMOVED=10
+)
+
+
+def generate_parser():
+    parser = argparse.ArgumentParser(
+        description="Checks the status of a MongoDB replica set",
+    )
+    parser.add_argument(
+        "-c", "--containers", required=True,
+        help='container name(s) for MongoDB pods separated by , (e.g.: "mongodb,mongodb-service")',
+    )
+    return parser
+
+
+check_mongodb_cmd = (
+    "bash", "-c",
+    'mongo 127.0.0.1/admin -u admin -p "$MONGODB_ADMIN_PASSWORD" --eval="print(JSON.stringify((rs.status())))" --quiet'
+)
+
+
+def parse_mongo_result(output):
+    try:
+        js = json.loads(output)
+        return int(js["myState"])
+    except:
+        return None
+
+
+def analize(status):
+    return {
+        REPLSET_STATUS.PRIMARY: nagios.OK,
+        REPLSET_STATUS.SECONDARY: nagios.OK,
+        REPLSET_STATUS.STARTUP: nagios.WARN,
+        REPLSET_STATUS.ARBITER: nagios.WARN,
+        REPLSET_STATUS.RECOVERING: nagios.CRIT,
+        REPLSET_STATUS.FATAL: nagios.CRIT,
+        REPLSET_STATUS.STARTUP2: nagios.CRIT,
+        REPLSET_STATUS.UNKNOWN: nagios.CRIT,
+        REPLSET_STATUS.DOWN: nagios.CRIT,
+        REPLSET_STATUS.ROLLBACK: nagios.CRIT,
+        REPLSET_STATUS.REMOVED: nagios.CRIT
+    }.get(status, nagios.UNKNOWN)
+
+
+def report(pods, rs_statuses, nag_statuses):
+    if not pods:
+        print "%s: Unable to locate any pods running mongodb" % (
+            nagios.status_code_to_label(nagios.UNKNOWN),)
```

Comment:
This is one funny bit of Python syntax, `()` is an empty tuple, `(x, y)` is a 2-element tuple, but `(p)` is not a tuple -- it is just a parenthesized expression. To write a literal 1-element tuple the comma is required: `(p,)`.